core/src/main/java/org/elasticsearch/action/percolate/TransportPercolateAction.java

```diff
@@ -196,9 +196,8 @@ public static SearchRequest createSearchRequest(PercolateRequest percolateReques
 
         PercolatorQueryBuilder percolatorQueryBuilder = new PercolatorQueryBuilder(percolateRequest.documentType(), documentSource);
         if (querySource != null) {
-            QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
-            queryParseContext.reset(XContentHelper.createParser(querySource));
-            queryParseContext.parseFieldMatcher(parseFieldMatcher);
+            QueryParseContext queryParseContext = new QueryParseContext(queryRegistry, XContentHelper.createParser(querySource),
```

Comment:
just noticed that below we have try finally around the parser creation but here we don't have it. Should we have that here too? who closes the parser after all?